narwhals/_compliant/group_by.py

```diff
@@ -108,20 +108,19 @@ def _temporary_name(key: str) -> str:
             key_str = str(key)  # pandas allows non-string column names :sob:
             return f"_{key_str}_tmp{'_' * (tmp_name_length - len(key_str) - 5)}"
 
-        output_names = _evaluate_aliases(compliant_frame, keys)
-
+        output_names_grouped = [expr._evaluate_aliases(compliant_frame) for expr in keys]
```

Comment:
If someone has a better variable name I am open to suggestions. Meaning is: this is not flattened